src/createClient.ts

```diff
@@ -0,0 +1,27 @@
+import type { SafeConfig, SafeConfigWithSigner } from '@/types/index.js'
+import { createSafeClient } from '@safe-global/sdk-starter-kit'
+
+const getPublicClientConfig = ({ provider, safeAddress, safeOptions }: SafeConfig) => ({
+  signer: undefined,
+  provider,
+  ...(safeAddress ? { safeAddress } : { safeOptions })
+})
+
+/**
+ * Creates a SafeClient instance with public method capabilities.
+ * @param config Config object for the Safe client
+ * @returns Safe client instance with public method capabilities
+ */
+export const createPublicClient = (config: SafeConfig) =>
+  createSafeClient(getPublicClientConfig(config))
+
+/**
+ * Creates a SafeClient instance with signer capabilities.
+ * @param config Config object for the Safe client with mandatory `signer` property
+ * @returns Safe client instance with signer capabilities
+ */
+export const createSignerClient = ({ signer, ...config }: SafeConfigWithSigner) =>
```

Comment:
it's not redundant to have public and signer clients instead a unique client that can have a signer ?